app/scripts/controllers/locationPickerMap.ctrl.js

```diff
@@ -113,11 +114,17 @@ angular.module('yourStlCourts').controller('LocationPickerMapCtrl', function ($s
   }
 
   function addMunicipality(municipality, mapMunicipalityId) {
-    updateUnincorporatedCount(municipality, true);
-    if(!_.find(ctrl.selectedMunicipalities, {id: municipality.id})) {
-      ctrl.selectedMunicipalities.push(municipality);
+    if (ctrl.selectedMunicipalities.length < 5) {
```

Comment:
Instead of returning true or false to the "addFunctionality" method, throw a new Error('error msg') from the method if your condition is hit and then above, just use try/catch. It would read more intuitively. Alternatively, you can create another method "canAddMunicipality" to check before doing but don't combine the two.